shared/src/business/useCaseHelper/serveDocumentAndGetPaperServicePdf.js

```diff
@@ -0,0 +1,63 @@
+const {
+  aggregatePartiesForService,
+} = require('../utilities/aggregatePartiesForService');
+const { saveFileAndGenerateUrl } = require('./saveFileAndGenerateUrl');
+
+/**
+ * serveDocumentAndGetPaperServicePdf
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @param {object} providers.caseEntity the case entity containing the docket entry to serve
+ * @param {string} providers.docketEntryId the id of the docket entry to serve
+ * @returns {Promise<*>} the updated case entity
+ */
+exports.serveDocumentAndGetPaperServicePdf = async ({
+  applicationContext,
+  caseEntity,
+  docketEntryId,
+}) => {
+  const servedParties = aggregatePartiesForService(caseEntity);
+
+  await applicationContext.getUseCaseHelpers().sendServedPartiesEmails({
+    applicationContext,
+    caseEntity,
+    docketEntryId,
+    servedParties,
+  });
+
+  if (servedParties.paper.length > 0) {
+    const { PDFDocument } = await applicationContext.getPdfLib();
```

Comment:
This gives me pause that we could be sending Email service before possibly adding coversheets or the system failing later on in the process. It would appear that this would require a significant amount of refactoring to adjust the order of operations. It is something we need to consider whenever we write code that involves service.

* get all of the ducks in a row
* then serve parties once the documents are ready.

Reply:
It looks like this is recycling code, and trying to consolidate it it in one place. 

I have created https://github.com/ustaxcourt/ef-cms/issues/901 to help address this concern.